internal/controller/driver_controller.go

```diff
@@ -436,6 +437,30 @@ func (r *driverReconcile) reconcileControllerPluginDeployment() error {
 			utils.LeaderElectionRetryPeriodContainerArg(leaderElectionSpec.RetryPeriod),
 		}
 
+		logRotator := &csiv1a1.LogRotationSpec{}
+		if r.driver.Spec.Log != nil {
+			logRotator = r.driver.Spec.Log.Rotation
+		}
+		var logRotationContainerArgs []string
+		var logRotationCsiAddonsSidecarContainerArgs []string
+		securityContext := &corev1.SecurityContext{Privileged: ptr.To(false)}
+		// choosing a different approach as need to check nil values
+		if logRotator != nil {
+			logRotationContainerArgs = []string{
+				utils.LogToStdErrContainerArg,
+				utils.AlsoLogToStdErrContainerArg,
+				utils.LogFileContainerArg(fmt.Sprintf("csi-%splugin", r.driverType)),
+			}
+			logRotationCsiAddonsSidecarContainerArgs = []string{
+				utils.LogToStdErrContainerArg,
+				utils.AlsoLogToStdErrContainerArg,
+				utils.LogFileContainerArg("csi-addons"),
+			}
+			securityContext = &corev1.SecurityContext{
+				Privileged: ptr.To(true),
+			}
```

Comment:
Possible to move this to helper to avoid duplication?

Reply:
Done